packages/paste-core/components/data-grid/__tests__/index.spec.tsx

```diff
@@ -68,64 +68,74 @@ describe('Data Grid', () => {
 
-      userEvent.keyboard('{arrowleft}');
+      await userEvent.keyboard('{arrowleft}');
       expect(headerCell).toHaveFocus();
 
-      userEvent.keyboard('{arrowdown}');
+      await userEvent.keyboard('{arrowdown}');
       expect(headerCell.getAttribute('tabindex')).toBe('-1');
       expect(firstInputCell).toHaveFocus();
       expect(firstInputCell.getAttribute('tabindex')).toBe('0');
 
-      userEvent.keyboard('{arrowup}');
+      await userEvent.keyboard('{arrowup}');
       expect(firstInputCell.getAttribute('tabindex')).toBe('-1');
       expect(headerCell.getAttribute('tabindex')).toBe('0');
       expect(headerCell).toHaveFocus();
 
       // TEST: toggles actionable mode with [enter] and [escape] keys
-      headerCell.focus();
+      act(() => {
+        headerCell.focus();
+      });
       expect(wrapper.getAttribute('data-actionable')).toBe('false');
-      userEvent.keyboard('{enter}');
+      await userEvent.keyboard('{enter}');
       expect(wrapper.getAttribute('data-actionable')).toBe('true');
-      userEvent.keyboard('{esc}');
+      await userEvent.keyboard('{esc}');
       expect(wrapper.getAttribute('data-actionable')).toBe('false');
 
       // TEST: should enable Actionable mode when clicking into the DataGrid
       expect(wrapper.getAttribute('data-actionable')).toBe('false');
-      userEvent.click(headerCell);
+
+      await userEvent.click(headerCell);
       expect(wrapper.getAttribute('data-actionable')).toBe('true');
 
       // TEST: should correctly tab through focusable elements in actionable mode
-      headerCell.focus();
+      act(() => {
+        headerCell.focus();
+      });
 
       // Down to input cell
-      userEvent.keyboard('{arrowdown}');
+      await userEvent.keyboard('{arrowdown}');
       // Swap to actionable
-      userEvent.keyboard('{enter}');
+      await userEvent.keyboard('{enter}');
+
       expect(firstRowFirstInputCell).toHaveFocus();
-      userEvent.tab();
+      await userEvent.tab();
       expect(firstRowSecondInputCell).toHaveFocus();
-      userEvent.tab();
-      userEvent.tab();
-      userEvent.tab();
+      await userEvent.tab();
+      await userEvent.tab();
+      await userEvent.tab();
       expect(secondRowFirstInputCell).toHaveFocus();
 
       /*
        * TEST: handles entering actionable mode from a cell without focusable children correctly
        * I added this particular sequence because it was a reproducable bug in my manual tests
        */
-      headerCell.focus();
-      userEvent.keyboard('{enter}');
-      userEvent.keyboard('{arrowdown}');
+      act(() => {
+        headerCell.focus();
+      });
+      await userEvent.keyboard('{enter}');
+      await userEvent.keyboard('{arrowdown}');
       expect(firstInputCell).toHaveFocus();
-      userEvent.keyboard('{enter}');
-      userEvent.tab();
+      await userEvent.keyboard('{enter}');
+      await userEvent.tab();
       expect(firstRowSecondInputCell).toHaveFocus();
-    });
+    }, 25000);
```

Comment:
I know this isn't ideal but this test kept timing out at around 20 seconds. I bumped it up to 25 and it's now passing.

Open to suggestions if someone can spot an alternative

Reply:
it looks like this test case is quite big and we're testing several conditions. would it be possible to split this test case in smaller ones?

Reply:
Same recommendation here, we should split this up imo

Reply:
Agreed, I'll split this out into smaller test